### PR TITLE
function code publish

### PR DESCRIPTION
**Pull request by uday31in**

### CODE REVIEW

**Comment by uday31in** [comment]
Deployments is successful 
image

Nested deployment is triggered as expected

image


However function is still running into an issue

image

@krnese  and @marvinbuss

File: infrastructure/terraform/main.tf [hunk 1]
```diff
@@ -2,14 +2,14 @@ provider "azurerm" {
...
-          recover_soft_deleted_keys                  = true
-          recover_soft_deleted_secrets               = true
+          purge_soft_delete_on_destroy               = true
+          purge_soft_deleted_certificates_on_destroy = true
+          purge_soft_deleted_keys_on_destroy         = true
+          purge_soft_deleted_secrets_on_destroy      = true
+          recover_soft_deleted_key_vaults            = false
+          recover_soft_deleted_certificates          = false
+          recover_soft_deleted_keys                  = false
+          recover_soft_deleted_secrets               = false
```
Comment (marvinbuss) [inline infrastructure/terraform/main.tf:12]: I would keep these to true to at least recover soft-deleted instances.

File: infrastructure/terraform/locals.tf [hunk 2]
```diff
@@ -23,6 +23,7 @@ locals {
   function_service_plan_name = "${local.prefix}-${var.function_name_suffix}"
   cleaned_function_storage_account_name= replace("${local.function_name}}", "/[^a-z0-9]/", "")
   function_storage_account_name = lower(substr(local.cleaned_function_storage_account_name, 0, min(length(local.cleaned_function_storage_account_name), 22)))
+  azure_function_name_helloworld = "${local.function_name}-${var.azure_function_helloworld_service_name_suffix}"
```
Comment (marvinbuss) [inline infrastructure/terraform/locals.tf:26]: Can we get rid of the sample hello world function?

File: infrastructure/terraform/deployment.tf [hunk 3]
```diff
@@ -89,18 +89,18 @@ module "data_factory" {
...
-  location = var.location
-  resource_group_name = azurerm_resource_group.ingestion.name
-  docintel_service_name = local.docintel_service_name
-}
+# module "document_intelligence" {
+#   source = "./modules/documentintel"
+#   location = var.location
+#   resource_group_name = azurerm_resource_group.ingestion.name
+#   docintel_service_name = local.docintel_service_name
+# }
```
Comment (marvinbuss) [inline infrastructure/terraform/deployment.tf:97]: Do you want to re-enable this or is this no longer required?

File: infrastructure/terraform/deployment.tf [hunk 4]
```diff
@@ -61,20 +61,20 @@ module "videoindexer" {
...
+#   search_service_name = local.azure_search_service_name
+#   sku = "standard"
+#   partition_count = var.partition_count
+#   replica_count = var.replica_count
+#   log_analytics_workspace_id = module.azure_log_analytics.log_analytics_id
+#   cmk_uai_id = module.azure_managed_identity.user_assigned_identity_id
+#   cmk_key_vault_id = module.azure_key_vault.key_vault_id
+#   subnet_id = var.subnet_id
+#   cmk_key_name = module.azure_key_vault.key_vault_cmk_name
+# }
```
Comment (marvinbuss) [inline infrastructure/terraform/deployment.tf:77]: Do you want to re-enable this or is this no longer required?

File: infrastructure/terraform/modules/aoai/aoai.tf [hunk 5]
```diff
@@ -25,7 +25,7 @@ resource "azurerm_cognitive_account" "cognitive_service" {
     trimsuffix(replace(var.key_vault_uri, "https://", ""), "/")
   ]
   kind               = var.cognitive_service_kind
-  local_auth_enabled = false
+  local_auth_enabled = true
```
Comment (marvinbuss) [inline infrastructure/terraform/modules/aoai/aoai.tf:28]: Are we now using key auth? Why is this required? Please disable if not required.

File: infrastructure/terraform/modules/functions/rbac-uai.tf [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+#   role_definition_name = "Cognitive Services OpenAI Contributor"
+#   principal_id         = data.azurerm_user_assigned_identity.user_assigned_identity.principal_id
+# }
+
+# # UAI - Assiging RBAC to OpenAI Account for functions to call the APIs - ShortClip
+# resource "azurerm_role_assignment" "uai_aoai_cognitive_service_openai_user" {
+#   scope                = var.cognitive_service_id
+#   role_definition_name = "Cognitive Services OpenAI User"
+#   principal_id         = data.azurerm_user_assigned_identity.user_assigned_identity.principal_id
+# }
```
Comment (marvinbuss) [inline infrastructure/terraform/modules/functions/rbac-uai.tf:40]: Please remove if this is no longer required.

File: infrastructure/terraform/modules/keyvault/kv.tf [hunk 7]
```diff
@@ -15,17 +15,17 @@ resource "azurerm_key_vault" "key_vault" {
     virtual_network_subnet_ids = []
   }
   public_network_access_enabled = true
-  purge_protection_enabled      = true
```
Comment (marvinbuss) [inline infrastructure/terraform/modules/keyvault/kv.tf:18]: Why are we removing this and change the provider config?
Is this solving anything?

**Review by marvinbuss — CHANGES_REQUESTED** [review]
Please review comments.

File: infrastructure/terraform/locals.tf (hunk 2 above)
Comment (uday31in) [reply infrastructure/terraform/locals.tf:26]: we dont have a consistent health check of the service so this helps for the time being to test the know good function to test infra deployment.

File: infrastructure/terraform/modules/functions/rbac-uai.tf (hunk 6 above)
Comment (uday31in) [reply infrastructure/terraform/modules/functions/rbac-uai.tf:40]: we want to revert to UAI asap. keeping this for time being.

**Review by marvinbuss — APPROVED** [review]
